model/role.go

```diff
@@ -579,68 +585,129 @@ func MakeDefaultRoles() map[string]*Role {
+		Permissions: []string{
+			PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_GROUPS.Id,
+			PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_TEAMS.Id,
+			PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_CHANNELS.Id,
+			PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_PERMISSIONS.Id,
+			PERMISSION_WRITE_SYSCONSOLE_USERMANAGEMENT_GROUPS.Id,
+			PERMISSION_WRITE_SYSCONSOLE_USERMANAGEMENT_TEAMS.Id,
+			PERMISSION_WRITE_SYSCONSOLE_USERMANAGEMENT_CHANNELS.Id,
+			PERMISSION_READ_SYSCONSOLE_AUTHENTICATION.Id,
+		},
+		SchemeManaged: false,
+		BuiltIn:       true,
+	}
+
+	roles[SYSTEM_READ_ONLY_ADMIN_ROLE_ID] = &Role{
+		Name:        "system_read_only_admin",
+		DisplayName: "authentication.roles.system_read_only_admin.name",
+		Description: "authentication.roles.system_read_only_admin.description",
+		Permissions: []string{
+			PERMISSION_READ_SYSCONSOLE_ABOUT.Id,
+			PERMISSION_READ_SYSCONSOLE_REPORTING.Id,
+			PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT.Id,
+			PERMISSION_READ_SYSCONSOLE_ENVIRONMENT.Id,
+			PERMISSION_READ_SYSCONSOLE_SITE.Id,
+			PERMISSION_READ_SYSCONSOLE_AUTHENTICATION.Id,
+			PERMISSION_READ_SYSCONSOLE_PLUGINS.Id,
+			PERMISSION_READ_SYSCONSOLE_INTEGRATIONS.Id,
+			PERMISSION_READ_SYSCONSOLE_EXPERIMENTAL.Id,
+		},
+		SchemeManaged: false,
+		BuiltIn:       true,
+	}
+
+	roles[SYSTEM_MANAGER_ROLE_ID] = &Role{
+		Name:        "system_manager",
+		DisplayName: "authentication.roles.system_manager.name",
+		Description: "authentication.roles.system_manager.description",
+		Permissions: []string{
+			PERMISSION_READ_SYSCONSOLE_ABOUT.Id,
+			PERMISSION_READ_SYSCONSOLE_REPORTING.Id,
+			PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_GROUPS.Id,
+			PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_TEAMS.Id,
+			PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_CHANNELS.Id,
+			PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_PERMISSIONS.Id,
+			PERMISSION_WRITE_SYSCONSOLE_USERMANAGEMENT_GROUPS.Id,
+			PERMISSION_WRITE_SYSCONSOLE_USERMANAGEMENT_TEAMS.Id,
+			PERMISSION_WRITE_SYSCONSOLE_USERMANAGEMENT_CHANNELS.Id,
+			PERMISSION_WRITE_SYSCONSOLE_USERMANAGEMENT_PERMISSIONS.Id,
+			PERMISSION_READ_SYSCONSOLE_ENVIRONMENT.Id,
+			PERMISSION_WRITE_SYSCONSOLE_ENVIRONMENT.Id,
+			PERMISSION_READ_SYSCONSOLE_SITE.Id,
+			PERMISSION_WRITE_SYSCONSOLE_SITE.Id,
+			PERMISSION_READ_SYSCONSOLE_AUTHENTICATION.Id,
+			PERMISSION_READ_SYSCONSOLE_PLUGINS.Id,
+			PERMISSION_READ_SYSCONSOLE_INTEGRATIONS.Id,
+			PERMISSION_WRITE_SYSCONSOLE_INTEGRATIONS.Id,
+		},
+		SchemeManaged: false,
+		BuiltIn:       true,
+	}
+
+	// SysconsoleAncillaryPermissions maps the non-sysconsole permissions required by each sysconsole permission
+	SysconsoleAncillaryPermissions := map[string][]*Permission{
+		PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_CHANNELS.Id: {
+			PERMISSION_READ_PUBLIC_CHANNEL,
+			PERMISSION_READ_CHANNEL,
+		},
+		// PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_USERS.Id:     {PERMISSION_EDIT_OTHER_USERS},
+		// PERMISSION_READ_SYSCONSOLE_ENVIRONMENT.Id:              {PERMISSION_MANAGE_JOBS},
```

Comment:
I'm not comfortable giving `PERMISSION_EDIT_OTHER_USERS` and `PERMISSION_MANAGE_JOBS` for read-only access to these sections, so I'm going to create some new permissions.